### ISSUE TITLE
implement conv1d cross platform without a dependency on TorchSharp

### ISSUE THREAD

**Issue by gsgou** [dotnet/machinelearning#6799]
**Is your feature request related to a problem? Please describe.**
I need to pre-process data before invoking a ONNX model in xamarin and .net maui.
Very often this requires conv1d but TorchSharp isn't available on mobile.
Do you plan to add some basic operations as conv1d in the lib soon?

[conv1d](https://pytorch.org/docs/stable/generated/torch.nn.functional.conv1d.html)

**Comment by GeorgeS2019** [comment]
@michaelgsharp
Please add deep-learning label to all TorchSharp/HuggingFace issues
![image](https://github.com/dotnet/machinelearning/assets/49812372/c4a04e8a-73ef-4046-9948-6db34d528b9d)
